src/routes/email/guards/email-update.guard.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  Injectable,
+} from '@nestjs/common';
+import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { verifyMessage } from 'viem';
+
+/**
+ * The EmailUpdateGuard guard should be used on routes that require
+ * authenticated actions on updating email addresses.
+ *
+ * This guard therefore validates if the message came from the specified account.
+ *
+ * The following message should be signed:
+ * email-update-${chainId}-${safe}-${emailAddress}-${account}-${timestamp}
+ *
+ * (where ${} represents placeholder values for the respective data)
+ *
+ * To use this guard, the route should have:
+ * - the 'chainId' declared as a parameter
+ * - the 'safeAddress' declared as a parameter
+ * - the 'emailAddress' as part of the JSON body (top level)
+ * - the 'account' as part of the JSON body (top level)
+ * - the 'signature' as part of the JSON body (top level) - see message format to be signed
+ * - the 'timestamp' as part of the JSON body (top level)
+ */
+@Injectable()
+export class EmailUpdateGuard implements CanActivate {
```

Comment:
I have created a new guard for now but would like to revisit this in a follow up PR to refactor the three into one.